File: Windows/Python/prime3a.py
#!/usr/bin/python3
#-----------------------------------------------------------------------------------------------------------------------
# prime3:   Generate the first 100 prime numbers.
#-----------------------------------------------------------------------------------------------------------------------

import sys

try:
    num_primes = int(sys.argv[1])
except (IndexError, ValueError):
    num_primes = 100
finally:
    if num_primes < 1:
        num_primes = 100

try:
    output_file = sys.argv[2]
    file = open(output_file, "w")
except IndexError:
    file = sys.stdout
except IOError:
    sys.stderr.write("Unable to open output file")
    file = sys.stdout

number = 1
format_string = "%20i: %20li\n"

file.write(format_string % (1, 2))

for count in range(2, 1 + num_primes):
    while True:    #This simulates a Do Loop
        number = 2 + number
        divisor = 3
        divisor_square = divisor * divisor

        while divisor_square <= number:
            remainder = number % divisor

            if remainder == 0:
                break
            else:
                divisor = 2 + divisor
                divisor_square = divisor * divisor
        else:
            file.write(format_string % (count, number))
            break
    if (count % 1000) == 0:
        file.flush()
        sys.stderr.write("%d primes found\n" % count)

sys.stderr.write("%d primes found\n" % count)
file.close()
sys.exit(0)

File: Windows/js/prime3a.js
function main() {
    var count;
    var divisor;
    var divisorsquare;
    var remainder;
    var isprime;
    var maxPrimes = 100;
    var outputFile = null;

    const output = require('fs');
    const sprintf = require('sprintf-js').sprintf;
    const args = process.argv.slice(2);

    try {
        maxPrimes = parseInt(args[0]);

        if (maxPrimes < 1 || isNaN(maxPrimes)) {
            maxPrimes = 100;
        }
    } catch (e) {
    }

    try {
        outputFile = args[1];

        if (!outputFile) {
            outputFile = null;
        }
    } catch (e) {
    }

    var outputLines = [];
    outputLines.push(sprintf("%20d: %20d", 1, 2));
    var number = 1;

    for (count = 2; count <= maxPrimes; count++) {
        do {
            number = 2 + number;
            divisor = 3;
            divisorsquare = divisor * divisor;
            isprime = true;
            while (isprime && divisorsquare <= number) {
                remainder = number % divisor;
                if (remainder == 0) {
                    isprime = false;
                } else {
                    divisor = 2 + divisor;
                    divisorsquare = divisor * divisor;
                }
            }
            if (isprime) {
                outputLines.push(sprintf("%20d: %20d", count, number));
            }
        } while (!isprime);

        if (count % 1000 == 0) {
            console.error(sprintf("%d primes found.", count));
        }
    }
    if (outputFile) {
        output.writeFileSync(outputFile, outputLines.join("\n"));
    } else {
        for (var i = 0; i < outputLines.length; i++) {
            console.log(outputLines[i]);
        }
    }
    console.error(sprintf("%d primes found.", count - 1));
}
main();
